src/docs/api/public.mdx

```diff
@@ -103,118 +109,230 @@ class OrganizationEventsEndpoint(...):
-  and pass the serializer as a parameter in `extend_schema`. See
-  [here](https://github.com/getsentry/sentry/blob/master/src/sentry/api/serializers/models/organization_member/scim.py)
+Specify the request serializer.
+
+- Request serializers, since they use DRF serializers, can generally just be the serializer itself.
+  There is plenty of customization and if needed can be overriden with `inline_serializer`. See
+  [here](https://github.com/getsentry/sentry/blob/b04c1c04f1e86bc040a671ad5b545d3b70132140/src/sentry/scim/endpoints/members.py#L434-L440)
   for an example of this.
-- If the response is a wrapper or you need more customization, you can use the `inline_sentry_response_serializer` function.
+- If the endpoint has no request body, you can either omit the `request` field or set it to `None`.
 
-Specify the endpoint description on the docs by adding a function docstring to the request.
 ```python
-from sentry.apidocs.utils import inline_sentry_response_serializer
+request=TeamPostSerializer,
+```
 
-@extend_schema(
-  responses={
-    200: inline_sentry_response_serializer(
-      "OrganizationList",
-      List[OrganizationSerializerResponse])
-      # A common use case is to return a list of serialized items
-  }
-)
+#### Success Responses
+
+Specify the return type of success responses, which are used to generate the response schema and 
+validate the sample response body. There are three ways to do this:
+
+1. If the success response is a single object instead of a list, you can pass a DRF serializer as
+  the response. In order for this serializer to generate a schema, its `serialize` method must be
+  typed to return a TypedDict. 
+
+  For example, this [sample code](https://github.com/getsentry/sentry/blob/0b9b6563bc4e232334cfc71f136a49117171d13f/src/sentry/scim/endpoints/members.py#L202)
+  has the `200` status code returning an `OrganizationMemberSCIMSerializer`. It's [`serialize`](https://github.com/getsentry/sentry/blob/0b9b6563bc4e232334cfc71f136a49117171d13f/src/sentry/api/serializers/models/organization_member/scim.py#L14-L16)
+  method is typed to return an [`OrganizationMemberSCIMSerializerResponse`](https://github.com/getsentry/sentry/blob/0b9b6563bc4e232334cfc71f136a49117171d13f/src/sentry/api/serializers/models/organization_member/response.py#L60)
+  TypedDict which specifies the typing of the response body.
+
+  Note that in order to have certain fields be optional, you must inherit from a TypedDict with
+  `total=False` in it's class header. Optional fields should be grouped together and cannot exist in
+  the same class as required fields.
+  See [here](https://github.com/getsentry/sentry/blob/0b9b6563bc4e232334cfc71f136a49117171d13f/src/sentry/api/serializers/models/organization_member/response.py#L53)
+  for an example of this.
+  ```python
+  class ExampleResponseOptional(TypedDict, total=False):
+      optionalStringField: str
```

Comment:
Is this actually optional? It doesn't have `Optional[str]`

Reply:
Is there a difference between optional field and fields that may be null?

Reply:
optional fields may or may not exist in the response. Fields that may be null must exist in the response, but they can be null

Reply:
A field can both be optional and null, so it may or may not exist in the response and can also be null when it is there

Reply:
Added `Note that a field can be both optional and null.` to the bottom of the section